third-party/cmake/DownloadAndExtractFile.cmake

```
@@ -59,41 +59,48 @@ string(REPLACE ";" "-" fname "${fname}")
 set(file ${DOWNLOAD_DIR}/${fname})
 message(STATUS "file: ${file}")
 
-message(STATUS "downloading...
-     src='${URL}'
-     dst='${file}'
-     timeout='${timeout_msg}'")
-
-file(DOWNLOAD ${URL} ${file}
-  ${timeout_args}
-  ${hash_args}
-  STATUS status
-  LOG log)
-
-list(GET status 0 status_code)
-list(GET status 1 status_string)
-
-if(NOT status_code EQUAL 0)
-  # Retry on certain errors, e.g. CURLE_COULDNT_RESOLVE_HOST, which is often
-  # seen with libtermkey (www.leonerd.org.uk).
-  if((status_code EQUAL 6)  # "Couldn't resolve host name"
-    OR (status_code EQUAL 7))  # "Couldn't connect to server"
-    message(STATUS "warning: retrying '${URL}' (${status_string}, status ${status_code})")
-    execute_process(COMMAND ${CMAKE_COMMAND} -E sleep 10)
-    file(DOWNLOAD ${URL} ${file}
-      ${timeout_args}
-      ${hash_args}
-      STATUS status
-      LOG log)
-    list(GET status 0 status_code)
-    list(GET status 1 status_string)
-  endif()
+set(EXISTING_SHA256 "")
+if(EXISTS ${file})
+  file(SHA256 ${file} EXISTING_SHA256)
+endif()
+
+if(NOT EXISTING_SHA256 STREQUAL ${EXPECTED_SHA256})
```

sorry i forgot to mention, is it possible that EXPECTED_SHA256 is empty string? then we should check something like:
```suggestion
if(${EXPECTED_SHA256} STREQUAL "" OR NOT EXISTING_SHA256 STREQUAL ${EXPECTED_SHA256})
```

---

> is it possible that EXPECTED_SHA256 is empty string?

Looks like no: https://github.com/neovim/neovim/blob/490615612ed5ec587c8023de28db495b3181de30/third-party/cmake/DownloadAndExtractFile.cmake#L13

---

seems EXPECTED_SHA256 can be set to 'SKIP' to drive some other logic, that's season I don't use it to compare with existing SHA256 in my first PR(although sha256 can't be 'skip')